tests/ui/consts/let-irrefutable-pattern-ice-120337.rs

```diff
@@ -1,9 +1,8 @@
-//@ check-pass
 #![feature(never_type)]
 #[derive(Copy, Clone)]
 pub enum E { A(!), }
 pub union U { u: (), e: E, }
-pub const C: () = {
+pub const C: () = { //~ ERROR evaluation of constant value failed
     let E::A(ref a) = unsafe { &(&U { u: () }).e};
 };
```

Comment:
Same remark regarding places vs expressions: there is no *value* of type `E` created here, so I don't think there should be any unreachable code reached.